turbopy/core.py

```diff
@@ -651,20 +651,27 @@ def set_value_from_keys(self, var_name, options):
         raise (KeyError("Grid configuration for " + var_name
                         + " not found."))
 
-    def generate_field(self, num_components=1):
+    def generate_field(self, num_components=1, placement_of_points="edge-centered"):
         """Returns squeezed :class:`numpy.ndarray` of zeros with
         dimensions :class:`Grid.num_points` and `num_components`.
 
         Parameters
         ----------
         num_components : int, defaults to 1
             Number of vector components at each point.
+        placement_of_points : str, defaults to "edge-centered"
+            Designate position of points on grid
         Returns
         -------
         :class:`numpy.ndarray`
             Squeezed array of zeros.
         """
-        return np.squeeze(np.zeros((self.num_points, num_components)))
+        amended_point_number = None
+        if placement_of_points == "edge-centered":
+            amended_point_number = self.num_points
+        else:
```

Comment:
I think I'd change this, to make "cell-centered" an explicit option. And maybe use the `else` clause to raise a `RuntimeError` (or maybe a `ValueError`) saying that an unknown placement option was specified.

Reply:
thank you, very good point (haha bad pun).